### ISSUE TITLE
Unable to work with Cinnamon under Mint 16

### ISSUE THREAD

**Issue by mbnoimi** [clearkimura/Customizer#37]
When I tried to use Customizer to customize Mint 16 x64 Cinnamon I got this error after running Xephyr
Full log: http://pastebin.com/KeDXBQnY
![xephyr on 9 0 ctrl shift grabs mouse and keyboard _001](https://f.cloud.github.com/assets/284547/1726611/44badd6c-6292-11e3-8f5e-0d6cf6ae8933.png)

How can I fix this issue?

P.S. Since 15 days I'm trying Customizer and I couldn't until now create any successful distro?! I'm very frustrated :(

**Comment by mbnoimi** [comment]
BTW, I tried to use Cinnamon and Cinnamon2D but I got same result

**Comment by ghost** [comment]
@mbnoimi 
First of all, Xephyr has few limitations. I, myself had seen some of them while test building few remixes (we don't call it distro because the term is misleading).

In your case, when running Cinnamon, most of its components failed to be loaded. In chroot environment, we cannot run some applications, such as Chromium web browser, Avant Window Navigator dock, Dbus daemons and such. This is most likely for security purposes, which we (end users) don't have control.

As far as I know, nobody has ever tried customizing Cinnamon, GNOME 3 and KDE yet.
And based on [Changes log](https://github.com/clearkimura/Customizer/wiki/Changes-log) wrote by the Developer, remastering Linux Mint ISO has more limitations than Ubuntu ISO.

In short, Linux Mint + Cinnamon might not be supported (based on your feedback).

Above all, I'm afraid we cannot provide any fix for this issue.
This is mainly because we don't have active developers working on this project at the moment. I am just a volunteering for documentation and small maintenance, so I could only try to reproduce any issues for clarification and suggest possible solution at my best knowledge.

I will leave this issue open, until this matter is clarified.

P.S.: I have created a form to collect feedback from Customizer users.
Please participate by sharing us your experience (whether success or failure).
You can find the link at [GitHub Wiki page](https://github.com/clearkimura/Customizer/wiki).

**Comment by mbnoimi** [comment]
> First of all, Xephyr has few limitations. I, myself had seen some of them while test building few remixes (we don't call it distro because the term is misleading).

Oops, How can I make desktop changes then?! Customizer allows me to install the extra packages but what about the configurations (ex. adding panels, wallpapers, icons... etc)

> As far as I know, nobody has ever tried customizing Cinnamon, GNOME 3 and KDE yet.

Wow, Mint is the most popular distro since years ago (see: http://distrowatch.com/) !!!

> Above all, I'm afraid we cannot provide any fix for this issue.
> This is mainly because we don't have active developers working on this project at the moment. I am just a volunteering for documentation and small maintenance, so I could only try to reproduce any issues for clarification and suggest possible solution at my best knowledge.

Thanks a lot for your contributions (we're all in open community). in the case you mentioned above it seems that Customizer is really dead because it doesn't support the most popular environments and hasn't active developing. I my self prefer to contribute but I don't have the suitable knowledge to start with it (I'm C++ developer I know nothing about Gambas).

> I have created a form to collect feedback from Customizer users.

I suggest to modify this page "I have created a form to collect feedback from Customizer users." because it hasn't the correct questions (ex. I can install packages but I can't customize the interface)

P.S. I appreciate your contributions a lot but I think this project need a revolutionary modifications because if the user have to modify anything manually that mean he/she don't need to use Customizer at all.

**Comment by fluxer** [comment]
Can you try with the latest version, from the gambas3 branch?

**Comment by ghost** [comment]
@mbnoimi 
What @fluxer means, the gambas3 package in 13.10 (that Linux Mint 16 uses) has version 3.1.1, which is likely outdated. Try using latest gambas3 from their PPA ( lookup at [their site](http://gambas.sourceforge.net) ), which is version 3.5.1.

And rest of comments as below.

> Oops, How can I make desktop changes then?! Customizer allows me to install the extra packages but what about the configurations (ex. adding panels, wallpapers, icons... etc)

Not sure about Cinnamon, but GNOME 2 allows those wallpapers and icons to be set via command-line as workaround. Then again, you may argue that this is not user friendly. But if GUI doesn't work, we have to rely on CLI sometimes.

> because it hasn't the correct questions (ex. I can install packages but I can't customize the interface)

Will revise the form again this weekend.

> P.S. I appreciate your contributions a lot but I think this project need a revolutionary modifications..

Not everyone has time to contribute at free cost (voluntary), because each of us has different living condition. 

Besides, I feel that it is quite ridiculous for a developer to support each and every "popular distro" (which may uses different structure of directories or compilation, etc.). 

We have hundreds of distro, because those each group of developers made easy for end users, so that users themselves don't have to customize at first place. Linux Mint is one of those.

>  because if the user have to modify anything manually that mean he/she don't need to use Customizer at all.

Whether you do it with Customizer or manual chroot, you are bound to do customization manually (to remove branding, etc.) and this is inevitable.

At the end of the day, open source projects rely on user feedback and contributions. If nobody gives feedback (as you are doing now) and nobody willing to contribute (as I am trying now), then any project will stagnate.

**Comment by fluxer** [comment]
@clearkimura What I mean is to try the latest version of Customizer ;). I I pushed some fixes to the gambas3 branch yesterday.

**Comment by ghost** [comment]
@fluxer 
I just read the commit log. My bad for misunderstanding.

@mbnoimi 
So please update to latest version of Customizer using the install.sh script.

And I have updated the form for question "Based on given information, which describes your experience?" with more specific options for the answer.

Kindly participate again after trying out with latest version of Customizer.

**Comment by mbnoimi** [comment]
@clearkimura 

> We have hundreds of distro, because those each group of developers made easy for end users, so that users themselves don't have to customize at first place. Linux Mint is one of those.

No, this is absolutely incorrect! We've just about 5 distros which they're the most popular distros while most of the others are derived from them. You can take the first 5 distros from http://distrowatch.com/ (just for example) thus you can support 75% of Linux distros.

@fluxer 

> I pushed some fixes to the gambas3 branch yesterday.

Thanks a lot, I'll test the new modifications tomorrow.

**Comment by mbnoimi** [comment]
@fluxer 

> wget https://dl.dropboxusercontent.com/u/54183088/install.sh
>  chmod +x install.sh
>  sudo ./install.sh -b gambas3 -i

I tried to install the new modifications but I still get same error message!

Do you've any idea how can I fix this issue? I've many modifications on the interface and I think most of them are hard to do by chroot so using Xephyr is very vital tool to me.

**Comment by fluxer** [comment]
@mbnoimi Can you try again? I do not have Ubuntu installed, but I have Xubuntu deployed on virtual machine which I will test this later. I have other issues on my setup with the nested X nightmare that's why I want you to test this.

@clearkimura Can you try the latest checkout of gambas3 branch as well?

**Comment by mbnoimi** [comment]
@fluxer 

> Can you try again? I do not have Ubuntu installed, but I have Xubuntu deployed on virtual machine which I will test this later. I have other issues on my setup with the nested X nightmare that's why I want you to test this.

Still this bug occurs :(

**Comment by fluxer** [comment]
@mbnoimi Dang it! I will play with it in virtual machine and see what I can do.

**Comment by fluxer** [comment]
@mbnoimi I've pushed fix, it works for Xubuntu and XFCE4 in nested X session but I have not tested it with Linux Mint and Cinnamon specificly. Give it another shot :)

**Comment by fluxer** [comment]
The issue is no more, I just tested it with Linux Mint 16 Cinnamon (i386).

**Comment by mbnoimi** [comment]
@fluxer 

Thanks a lot... it works like the charm.

**Comment by ghost** [comment]
@mbnoimi 
Glad to hear that this issue was solved.

@fluxer 
Many thanks for pushing the fix!

Last weekend, I just managed to download Linux Mint 13 ( based on 12.04 LTS ) Cinnamon 32-bit ISO, but didn't have enough time to try out latest gambas3 port as you requested.
Apologies.
Will try again by end of this month, just to double-confirm the status.

**Comment by ghost** [comment]
@mbnoimi 
@fluxer 
I have just done quick check on latest Customizer (gambas3 port), installed on 12.04 host system and loaded with Linux Mint 13 (based on 12.04 LTS) Cinnamon 32-bit.

For 'Desktop' feature, I've tested with all GUI sessions: Cinnamon, gnome-classic, gnome-fallback. All works. The issue is no more, as both of you claimed.

**Comment by fluxer** [comment]
@clearkimura Thanks for testing!
